### PR TITLE
cake 5: Add native type hints for Controller properties

### PR DESCRIPTION
**Pull request by othercorey**
Seems like a reasonable place to add them.

### CODE REVIEW

File: src/Controller/Controller.php [hunk 1]
```diff
@@ -106,7 +106,7 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      *
      * @var string
      */
-    protected $name;
+    protected string $name = '';
```
Comment (othercorey) [inline src/Controller/Controller.php:109]: This is to avoid a `?string` type because psalm complains when you check if initialized with `isset()`.

**Comment by ADmad** [comment]
There are a lot of places where FQCN has been used for the type which need to be replaced with just class/interface name along with `use` statements.

**Comment by othercorey** [comment]
> There are a lot of places where FQCN has been used for the type which need to be replaced with just class/interface name along with `use` statements.

I forgot to enable this sniff. Should we enable it by default?

**Comment by othercorey** [comment]
Fixed type hint namespaces.

File: src/Controller/Controller.php [hunk 2]
```diff
@@ -196,11 +196,14 @@ public function __construct(
     ) {
         if ($name !== null) {
             $this->name = $name;
-        } elseif ($this->name === null && $request) {
-            $this->name = $request->getParam('controller');
+        } elseif (!$this->name && $request) {
+            $controller = $request->getParam('controller');
+            if ($controller) {
+                $this->name = $request->getParam('controller');
```
Comment (markstory) [inline src/Controller/Controller.php:202]:
```suggestion
                $this->name = $controller;
```

Comment (othercorey) [reply src/Controller/Controller.php:202]: whoops, too many undo/redo.

File: tests/test_app/TestApp/Controller/ViewPostsController.php [hunk 3]
```diff
@@ -24,7 +24,7 @@ class ViewPostsController extends Controller
      *
      * @var string
      */
-    public $name = 'Posts';
+    public string $name = 'Posts';
```
Comment (ADmad) [inline tests/test_app/TestApp/Controller/ViewPostsController.php:27]:
```suggestion
    protected string $name = 'Posts';
```

**Pull request merged**